File: StudentCredit.Application/Applications/ApplicationsFive/Dtos/DividendReportSearchDto.cs
using StudentCredit.Data.Applications.ApplicationFive.Enums;
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Applications.ApplicationsFive.Dtos
{
	public class DividendReportSearchDto
	{
		public NomenclatureDto Bank { get; set; }

		public NomenclatureDto Year { get; set; }

		public ApplicationFiveType ApplicationFiveType { get; set; }

		public DateTime? From { get; set; }

		public DateTime? To { get; set; }

		public YearPeriod? Period { get; set; }
	}
}

File: WebApplication/src/modules/application/application-five/models/dividend-report-search.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { ApplicationFiveType } from "../enums/application-five-type.enum";
import { YearPeriod } from "../enums/year-period.enum";

export class DividendReportSearchDto {
  bank: NomenclatureDto;
  applicationFiveType: ApplicationFiveType;
  year: NomenclatureDto;
  from: Date;
  to: Date;
  period: YearPeriod;
}